src/workerd/api/tests/navigator-beacon-test.js

```diff
@@ -0,0 +1,29 @@
+import {
+  ok,
+  strictEqual,
+} from 'node:assert';
+
+const enc = new TextEncoder();
+
+// This one returns false with no error thrown because we're not in an IoContext
+ok(!navigator.sendBeacon('http://example.org', 'does not work'));
+
+export const navigatorBeaconTest = {
+  async test(ctrl, env, ctx) {
+    ok(navigator.sendBeacon('http://example.org', 'beacon'));
```

Comment:
Should we consider implementing our own local server to avoid any flakiness coming to our tests due to network issues?

Reply:
Actually, that reminds me, I meant to have the worker config here loop back on itself

Reply:
oh wait, hah, I already do, lol. If you notice in the wd-test I route the `internet` service back to `.worker`, which means this fetch will loop back to the exported fetch handler here. There should be no network flakiness possible here.